internal/validations/clusterlogforwarder/validate_clusterlogforwarderspec.go

```diff
@@ -204,12 +204,22 @@ func verifyInputs(spec *loggingv1.ClusterLogForwarderSpec, status *loggingv1.Clu
 			badInput("inputspec cannot have a negative limit threshold")
 		case input.Receiver != nil && !extras[constants.VectorName]:
 			badInput("ReceiverSpecs are only supported for the vector log collector")
-		case input.Receiver != nil && input.Receiver.HTTP == nil:
-			badInput("ReceiverSpec must define an HTTP receiver")
-		case isHTTPReceiver(input) && input.Receiver.HTTP.Format != loggingv1.FormatKubeAPIAudit:
-			badInput("invalid format specified for HTTP receiver")
-		case isHTTPReceiver(input) && !validPort(input.Receiver.HTTP.Port):
+		case input.Receiver != nil && input.Receiver.ReceiverTypeSpec == nil:
```

Comment:
Instead of moving them into a separate file, I moved them to a separate func in the same file: validateReceivers, which will get validated just after Inputs.

Reply:
I have been trying nad this implies a lot of changes. Could we consider this validation re-org as candidate for its own PR?

Reply:
This entire validation should be refactored.  I was thinking the 'receiver' validations could be moved out, but if the work is too invasive, please ignore this comment.